src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,455 @@
+  org_info = nothing
+  for row in srs_result
+    org_info = row
+    break
+  end
+  
+  if isnothing(org_info)
+    return Cartesian{NoDatum}
+  end
+  
+  # Handle different CRS organizations
+  org = ismissing(org_info.organization) ? "" : uppercase(org_info.organization)
+  coord_id = org_info.organization_coordsys_id
+  
+  if org == "EPSG"
+    return CoordRefSystems.get(EPSG{coord_id})
+  elseif org == "ESRI"
+    return CoordRefSystems.get(ESRI{coord_id})
+  elseif org == "NONE" || org == ""
+    # NONE organization indicates undefined/custom CRS
+    return Cartesian{NoDatum}
+  else
+    # Other organizations (custom authorities) - use Cartesian as fallback
+    # This branch is reachable for custom organization names
+    return Cartesian{NoDatum}
+  end
+end
+
+function gpkgreadtable(table_result, geom_column::String)
+  # Get the first row to determine structure
+  row_iter = Tables.rows(table_result)
+  first_row = nothing
+  for row in row_iter
+    first_row = row
+    break
+  end
+  
+  if isnothing(first_row)
+    return nothing
+  end
+  
+  # Get attribute names from the first row (geometry column already excluded)
+  attr_names = propertynames(first_row)
+  if isempty(attr_names)
+    return nothing
+  end
+  
+  # Collect all rows in a single pass
+  rows = Vector{Vector{Any}}()
+  for row in Tables.rows(table_result)
+    push!(rows, [getproperty(row, name) for name in attr_names])
+  end
+  
+  if isempty(rows)
+    return nothing
+  end
+  
+  # Create a named tuple of vectors (Tables.jl compatible)
+  columns = NamedTuple{Tuple(attr_names)}([
+    [row[i] for row in rows] for i in 1:length(attr_names)
+  ])
+  
+  return columns
+end
+
+function gpkgreadgeoms(data_result, geom_column::String, crs_type)
+  geometries = Geometry[]
+  geom_sym = Symbol(geom_column)
+  
+  for row in data_result
+    blob = getproperty(row, geom_sym)
+    if !ismissing(blob) && !isnothing(blob)
+      geom, _ = parse_gpb(blob, crs_type)
+      if !isnothing(geom)
+        push!(geometries, geom)
+      end
+    end
+  end
+  
+  return geometries
+end
+
+function gpkgmetadata(db::SQLite.DB, layer::Int)
+  # Get feature table names
+  contents_result = DBInterface.execute(db,
+    """
+    SELECT table_name, data_type
+    FROM gpkg_contents
+    WHERE data_type = 'features'
+    ORDER BY table_name
+    """)
+  contents = [row.table_name for row in contents_result]
+  
+  if isempty(contents)
+    error("No vector feature tables found in GeoPackage")
+  end
+  
+  if layer > length(contents)
+    error("Layer $layer not found. GeoPackage contains $(length(contents)) feature layers")
+  end
+  
+  table_name = contents[layer]
+  
+  # Get geometry column information
+  geom_result = DBInterface.execute(db,
+    """
+    SELECT column_name, srs_id
+    FROM gpkg_geometry_columns
+    WHERE table_name = ?
+    """, [table_name])
+  geom_info = first(geom_result)
+  geom_column = geom_info.column_name
+  srs_id = Int32(geom_info.srs_id)
```

Comment:
Is this explicit conversion to `Int32` necessary? Isn't the type stored correctly in the database?